flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -513,6 +519,56 @@ public static Configuration cloneConfiguration(Configuration configuration) {
 		return clonedConfiguration;
 	}
 
+	/**
+	 * Format the default gc logging options
+	 * @param logDirectory to save the gc log
+	 * @param conf flink configuration
+	 * @return the formatted gc logging options string
+	 */
+	public static String getGCLoggingOpts(String logDirectory, Configuration conf) {
+		boolean enableGCLogging = conf.getBoolean(CoreOptions.FLINK_JVM_DEFAULT_GC_LOGGING);
+		if (enableGCLogging) {
```

Comment:
I would move this check out of this method to simplify it. Then one would also not need to pass in a `Configuration` object.

Reply:
The check could then happen in `getTaskManagerShellCommand` or in another method `Optional<String> maybeGetGCLoggingOpts`

Reply:
Done